logging start of torch elastic workers.

aschhabra opened the pull request:
Summary:
We would like to log start of the workers. It will help with complete logging.


Test Plan:
unit tests


https://www.internalfb.com/intern/testinfra/testrun/6473924724652056

e2e tests
https://www.internalfb.com/mlhub/pipelines/runs/mast/f712311762-27449483648-TrainingApplication_V403K?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

Reviewed By: tnykiel

Differential Revision: D72297314




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

d4l3k reviewed (approved):
Would be nice to add a test for this in https://github.com/pytorch/pytorch/blob/6a6e073928ea2de04bb039d6940d07ab01060a5f/test/distributed/elastic/agent/server/test/api_test.py

aschhabra committed:
logging start of torch elastic workers (#150849)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/150849

We would like to log start of the workers. It will help with complete logging.

Also adding local_rank to metadata for end events workers.

Test Plan:
unit tests

```
buck test caffe2/test/distributed/elastic/agent/server/test:api_test
```
https://www.internalfb.com/intern/testinfra/testrun/10977524171554657

https://www.internalfb.com/mlhub/pipelines/runs/mast/f706385944-TrainingApplication_HSHWL?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

https://www.internalfb.com/mlhub/pipelines/runs/mast/f712311762-27449483648-TrainingApplication_P5UA6?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

https://fburl.com/scuba/pytorch_elastic_status/n3v2w9r4
 {F1977013861}

Reviewed By: kiukchung, d4l3k

Differential Revision: D72297314
